zebra-network/src/peer_set/inventory_registry.rs

```diff
@@ -111,9 +215,14 @@ impl InventoryRegistry {
         Ok(())
     }
 
-    /// Record that the given inventory `hash` is available from the peer `addr`
-    fn register(&mut self, hash: InventoryHash, addr: SocketAddr) {
-        self.current.entry(hash).or_default().insert(addr);
+    /// Record the given inventory `change` for the peer `addr`.
+    fn register(&mut self, change: InventoryChange) {
+        let status = change.as_ref().map(|_| ());
+        let (invs, addr) = change.inner();
```

Comment:
Optional: I was wondering if this would be easier to read if it's written in a way like:

```suggestion
        let (invs, addr, status) = change.into_parts();
```

I think you have to have this method implemented for `InventoryChange`:

```rust
impl InventoryChange {
    pub fn into_parts(self) -> (AtLeastOne<InventoryHash>, SocketAddr, InventoryMarker) {
        match self {
            Advertised((items, address)) => (items, address, Advertised(())),
            Missing((items, address)) => (items, address, Missing(())),
        }
    }
}
```

Reply:
I went with ad5967175 instead, which creates an `InventoryStatus::marker() -> InventoryMarker` method.